lib/Sema/DerivedConformanceCodable.cpp

```diff
@@ -89,6 +89,12 @@ static CodableConformanceType typeConformsToCodable(TypeChecker &tc,
   // Some generic types need to be introspected to get at their "true" Codable
   // conformance.
   auto canType = target->getCanonicalType();
+  if (auto referenceType = dyn_cast<ReferenceStorageType>(canType)) {
```

Comment:
Looks like no, since `getReferentType()` simply returns a `Type` when I need a `CanType`. If I `getAs`, what guarantees do I have about the referent type? Is the point here that it'll also be canonical?